### PR TITLE
Adds step to dynamically create opencontainers labels

### PR DESCRIPTION
**Pull request by meik99**
# Description

Modifies the GitHub CI tasks creating the operator image to include the docker/metadata-action.
This action adds standardized opencontainer labels.

## How can this be tested?

* Use `scopeo` to inspect the snapshot image built with this PR
* Check that all necessary labels exist in the manifest
* Check that the labels values are correct
## Checklist
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: hack/build/ci/prepare-build-variables.sh [hunk 1]
```diff
@@ -18,6 +18,8 @@ createDockerImageLabels() {
   if [[ "${GITHUB_REF_TYPE}" != "tag" ]] && [[ ! "${GITHUB_REF_NAME}" =~ ^release-* ]]; then
     echo "quay.expires-after=10d"
   fi
+
+  echo "build-date=$(date --iso-8601)"
```
[the remark this answers is not in the document]
Comment (meik99) [reply hack/build/ci/prepare-build-variables.sh:22]: yes

[the remark this answers is not in the document]
Comment (meik99) [reply hack/build/ci/prepare-build-variables.sh:22]: output of `skopeo inspect`
```
"Labels": {
        "architecture": "x86_64",
        "build-date": "2022-09-13T17:50:09.610067",
        "changelog-url": "https://github.com/Dynatrace/dynatrace-operator/releases",
        "com.redhat.build-host": "cpt-1008.osbs.prod.upshift.rdu2.redhat.com",
        "com.redhat.component": "ubi9-micro-container",
        "com.redhat.license_terms": "https://www.redhat.com/en/about/red-hat-end-user-license-agreements#UBI",
        "description": "ActiveGate works as a proxy between Dynatrace OneAgent and Dynatrace Cluster",
        "distribution-scope": "public",
        "io.k8s.description": "Dynatrace Operator image.",
        "io.k8s.display-name": "Dynatrace Operator",
        "io.openshift.expose-services": "",
        "io.openshift.tags": "dynatrace-operator",
        "maintainer": "Dynatrace LLC",
        "name": "Dynatrace Operator",
        "org.opencontainers.image.created": "2022-10-25T15:13:43.555Z",
        "org.opencontainers.image.description": "Kubernetes / OpenShift operator to manage Dynatrace OneAgent and ActiveGate lifecycle",
        "org.opencontainers.image.licenses": "Apache-2.0",
        "org.opencontainers.image.revision": "97d6fa6e38813dffc6ee24731d6681ab2a90341b",
        "org.opencontainers.image.source": "https://github.com/Dynatrace/dynatrace-operator",
        "org.opencontainers.image.title": "dynatrace-operator",
        "org.opencontainers.image.url": "https://github.com/Dynatrace/dynatrace-operator",
        "org.opencontainers.image.version": "snapshot-ci-dynamic-labels",
        "quay.expires-after": "10d",
        "release": "1",
        "summary": "Dynatrace is an all-in-one, zero-config monitoring platform designed by and for cloud natives. It is powered by artificial intelligence that identifies performance problems and pinpoints their root causes in seconds.",
        "url": "https://www.dynatrace.com",
        "vcs-ref": "97d6fa6e38813dffc6ee24731d6681ab2a90341b",
        "vcs-type": "git",
        "vcs-url": "https://github.com/Dynatrace/dynatrace-operator.git",
        "vendor": "Dynatrace LLC",
        "version": "1.x"
    },

```
`build-date` as well as `quay.expires-after` are present

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -95,6 +95,15 @@ jobs:
         id: prep
         run: |
           hack/build/ci/prepare-build-variables.sh
+      - name: Docker metadata
+        uses: docker/metadata-action@57396166ad8aefe6098280995947635806a0e6ea # v4.1.1
+        id: meta
+        with:
+          images: dynatrace-operator
```
[the remark this answers is not in the document]
Comment (meik99) [reply .github/workflows/ci.yaml]: Added a list of image names

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -95,6 +95,19 @@ jobs:
...
+      - name: Docker metadata
+        uses: docker/metadata-action@57396166ad8aefe6098280995947635806a0e6ea # v4.1.1
+        id: meta
+        with:
+          tags: ${{ steps.prep.outputs.docker_image_tag }}
+          images: |
+            dynatrace/dynatrace-operator
+            quay.io/dynatrace/dynatrace-operator
+            registry.connect.redhat.com/dynatrace/dynatrace-operator
+            gcr.io/dynatrace-marketplace-prod/dynatrace-operator
```
Comment (chrismuellner) [inline .github/workflows/ci.yaml:107]: unused, can be removed

**Review by chrismuellner — CHANGES_REQUESTED** [review]

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**